### PR TITLE
Export routerReducer for custom root reducers (eg. redux-loop)

### PR DESCRIPTION
**Pull request by D1plo1d**
Refactored `routerReducer` as a export of `connected-react-router` so that it can be used inside `redux-loop`'s `combineReducers` function.

### redux-loop example (with the new routerReducer export)

```javascript
import {
  install as installReduxLoop,
  combineReducers,
} from 'redux-loop'

import { createHashHistory } from 'history'
import { routerReducer, routerMiddleware } from 'connected-react-router'

const rootReducer = combineReducers({
  // ...your other reducers here
  router: routerReducer(history),
})

const enhancer = compose(
  applyMiddleware(
    routerMiddleware(history),
    // ... other middlewares ...
  ),
  installReduxLoop(),
)

const store = createStore(
  rootReducer,
  initialState,
  enhancer,
)
```

TL;DR: redux-loop support. All test pass. Previous functionality should not be effected.

### CODE REVIEW

**Comment by D1plo1d** [comment]
I didn't update the docs but IMHO I'd prefer to see  `router: routerReducer(history)` documented as the default since it is consistent with other redux libraries (eg. redux-form) and can be reasonably expected to be compatible with any other redux middleware/enhancer/higher-order-reducer library (eg. redux-loop) since it is consistent with the basic redux reducer pattern.

**Comment by ee0pdt** [comment]
Looks like this could be combined with the proposed solution to this issue: https://github.com/supasate/connected-react-router/issues/133
